core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -141,8 +141,8 @@ public CallOptions withRequestKey(@Nullable RequestKey requestKey) {
    */
   @ExperimentalApi
   @Nullable
-  public RequestKey getRequestKey() {
-    return requestKey;
+  public Attributes getAttributes() {
```

Comment:
`withAffinity()` SGTM. You may also want to use `affinity` instead of `attr` for the argument.